api/src/api/userRouter.test.ts

```diff
@@ -343,6 +345,186 @@ describe("userRouter", () => {
+              needsNexusDbaName: false,
+            }),
+            formationData: generateFormationData({
+              dbaBusinessNameAvailability: generateBusinessNameAvailability({
+                status: "AVAILABLE",
+                lastUpdatedTimeStamp: getCurrentDate()
+                  .subtract(1, "hour")
+                  .subtract(1, "minute")
+                  .toISOString(),
+              }),
+            }),
+          });
+          stubUserDataClient.get.mockResolvedValue(userData);
+
+          await request(app).get(`/users/123`).set("Authorization", "Bearer user-123-token");
+          expect(stubTimeStampBusinessSearch.search).not.toHaveBeenCalled();
+        });
+
+        it("updates businessNameAvailability and sets needsNexusDbaName", async () => {
+          const userData = generateUserData({
+            profileData: generateProfileData({
+              businessPersona: "FOREIGN",
+              needsNexusDbaName: false,
+            }),
+            formationData: generateFormationData({
+              businessNameAvailability: generateBusinessNameAvailability({
+                status: "AVAILABLE",
+                lastUpdatedTimeStamp: getCurrentDate()
+                  .subtract(1, "hour")
+                  .subtract(1, "minute")
+                  .toISOString(),
+              }),
+            }),
+          });
+          stubUserDataClient.get.mockResolvedValue(userData);
+          stubTimeStampBusinessSearch.search.mockResolvedValue(
+            generateBusinessNameAvailability({
+              status: "UNAVAILABLE",
+              similarNames: ["random-name"],
+            })
+          );
+
+          const result = await request(app).get(`/users/123`).set("Authorization", "Bearer user-123-token");
+          expect(stubTimeStampBusinessSearch.search).toHaveBeenCalled();
+          expect(result.body.formationData.businessNameAvailability.status).toEqual("UNAVAILABLE");
+          expect(result.body.profileData.needsNexusDbaName).toEqual(true);
+        });
+
+        it("only updates businessNameAvailability if needsNexusDbaName is false", async () => {
+          const userData = generateUserData({
+            profileData: generateProfileData({
+              businessPersona: "FOREIGN",
+              needsNexusDbaName: false,
+            }),
+            formationData: generateFormationData({
+              businessNameAvailability: generateBusinessNameAvailability({
+                status: "AVAILABLE",
+                lastUpdatedTimeStamp: getCurrentDate()
+                  .subtract(1, "hour")
+                  .subtract(1, "minute")
+                  .toISOString(),
+              }),
+              dbaBusinessNameAvailability: generateBusinessNameAvailability({
+                status: "AVAILABLE",
+                lastUpdatedTimeStamp: getCurrentDate()
+                  .subtract(1, "hour")
+                  .subtract(1, "minute")
+                  .toISOString(),
```

Comment:
```
getCurrentDate()
  .subtract(1, "hour")
  .subtract(1, "minute")
  .toISOString(),
```

is used all over the place and seems like a great opportunity for an extract function call